src/System.Private.CoreLib/shared/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -214,6 +214,74 @@ protected unsafe void Initialize(byte* pointer, long length, long capacity, File
         /// </summary>
         public override bool CanWrite => _isOpen && (_access & FileAccess.Write) != 0;
 
+        /// <summary>
+        /// Calls the given callback with a span of the memory stream data
+        /// </summary>
+        /// <param name="callback">the callback to be called</param>
+        /// <param name="state">A user-defined state, passed to the callback</param>
+        /// <param name="bufferSize">the maximum size of the memory span</param>
+        public override void CopyTo(Buffers.ReadOnlySpanAction<byte, object?> callback, object? state, int bufferSize)
+        {
+            // If we have been inherited into a subclass, the following implementation could be incorrect
+            // since it does not call through to Read() which a subclass might have overridden.
+            // To be safe we will only use this implementation in cases where we know it is safe to do so,
+            // and delegate to our base class (which will call into Read) when we are not sure.
+            if (GetType() != typeof(UnmanagedMemoryStream))
+            {
+                base.CopyTo(callback, state, bufferSize);
+                return;
+            }
+
+            if (callback == null) throw new ArgumentNullException(nameof(callback));
+
+            EnsureNotClosed();
+            EnsureReadable();
+
+            // Use a local variable to avoid a race where another thread
+            // changes our position after we decide we can read some bytes.
+            long pos = Interlocked.Read(ref _position);
+            long len = Interlocked.Read(ref _length);
+            long n = len - pos;
+            if (n <= 0)
+            {
+                return;
+            }
+
+            int nInt = (int)n; // Safe because n <= count, which is an Int32
+            if (nInt < 0)
+            {
+                return;  // _position could be beyond EOF
+            }
+            unsafe
+            {
+                if (_buffer != null)
+                {
+                    byte* pointer = null;
+
+                    RuntimeHelpers.PrepareConstrainedRegions();
```

Comment:
@jkotas, are the PrepareConstrainedRegions in this file still valuable for anything?